src/iop/highlights.c

```diff
@@ -439,28 +447,28 @@ void tiling_callback(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t
   if(d->mode == DT_IOP_HIGHLIGHTS_LAPLACIAN && filters && filters != 9u)
   {
     // Bayer CFA and guided laplacian method : prepare for wavelets decomposition.
-    const float scale = fmaxf(piece->iscale / roi_in->scale, 1.f);
+    const float scale = fmaxf(DS_FACTOR * piece->iscale / roi_in->scale, 1.f);
     const float final_radius = (float)((int)(1 << d->scales)) / scale;
-    const int scales = CLAMP((int)round(log2f(final_radius)), 0, MAX_NUM_SCALES);
+    const int scales = CLAMP((int)ceilf(log2f(final_radius)), 1, MAX_NUM_SCALES);
     const int max_filter_radius = (1 << scales);
 
     // Warning : in and out are single-channel in RAW mode
-    // in + out + interpolated + tmp + 2 * LF + HF + mask
+    // in + out + interpolated + ds_interpolated + ds_tmp + 2 * ds_LF + ds_HF + mask + ds_mask
     if(filters) // RAW
     {
-      tiling->factor = 2.f + 6.f * 4;
-      tiling->factor_cl = 2.f + 6.f * 4;
+      tiling->factor = 2.f + 2.f * 4 + 6.f * 4 / DS_FACTOR;
+      tiling->factor_cl =  2.f + 3.f * 4 + 5.f * 4 / DS_FACTOR;
```

Comment:
If I'm reading things right, all of the factor= and factor_cl= lines in tiling_callback should be using the square of DS_FACTOR as the divisor, since DS_FACTOR is the linear downsampling (so the 2D buffers are only 1/16 as large for DS_FACTOR==4).  The only harm done is smaller tiles.

Reply:
I think so too.